### PR TITLE
refactor(tree-wide): add & use calver implementation

### PR DESCRIPTION
**Pull request by uncenter**
- [x] Implement calver bump script
- [x] Add CI to auto bump modified styles on PR merge
- [x] Update documentation / PR templates / etc to remove mentions of semver/other versioning.
- [x] Prohibit bump script outside of CI (env.CI var check?)
- [x] Double check calver support with Stylus

The script works as-is already, see the above todo list for the remaining tasks before we can make this a reality.

### CODE REVIEW

**Comment by uncenter** [comment]
Whoops

**Review by NeonGamerBot-QK — DISMISSED** [review]

**Comment by NeonGamerBot-QK** [comment]
lmao

**Comment by uncenter** [comment]
No need to approve, thanks though! This PR introduces a new versioning system based on https://calver.org/, as we've realized Sem(antic)Ver(sioning) isn't proving useful or worth the time. In the future, the CI/GitHub Actions will automatically bump userstyle versions after PRs are merged, so worrying about versioning is no longer a concern of maintainers!

**Review by isabelroses — DISMISSED** [review]

File: scripts/bump-version/main.ts
Comment (sgoudham) [inline scripts/bump-version/main.ts:11]: Bit of a nit but we should prefer being direct in all documentation / try our best to be. Thoughts on the wording below?

```suggestion
    "This script is intended to be used in CI. Userstyle versions are automatically bumped after pull requests are merged.",
```

**Comment by sgoudham** [comment]
Wish we did this when starting out but hindsight is 20/20. Happy to approve once you've tested Stylus doesn't break with this new format.

File: scripts/bump-version/main.ts
Comment (uncenter) [reply scripts/bump-version/main.ts:11]: 100%, this reads much better. Thanks!

**Review by sgoudham — APPROVED** [review]

**Review by TadoTheMiner — APPROVED** [review]

**Review by sofiedotcafe — APPROVED** [review]

**Pull request merged**
